<a href="https://colab.research.google.com/github/Tclack88/misc-analysis/blob/main/mandarin_frequency_analysis/freq_overlap_mandarin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# sources:  refold: https://drive.google.com/drive/folders/1HyLEQlqm3IJ3tREVAw3n97kMtDCQCb_D?usp=sharing
#           heisig: https://docs.google.com/spreadsheets/d/1azf6QcVXt_UMXJJPo06TXNLDli4BL7hbUfPO83cF7HA/htmlview
#           frequency: https://lingua.mtsu.edu/chinese-computing/statistics/  # not used. Wiki is used instead
refold_data = "https://raw.githubusercontent.com/Tclack88/misc-analysis/main/mandarin_frequency_analysis/refold_1k.txt"
heisig_data1 = "https://raw.githubusercontent.com/Tclack88/misc-analysis/main/mandarin_frequency_analysis/heisig1.csv"
heisig_data2 = "https://raw.githubusercontent.com/Tclack88/misc-analysis/main/mandarin_frequency_analysis/heisig2.csv"
frequency_data = "https://raw.githubusercontent.com/Tclack88/misc-analysis/main/mandarin_frequency_analysis/charfreq_modern.csv"

In [2]:
import pandas as pd

# Frequency lists (words and chars)

## refold (words)
first 1000 HSK

In [3]:
refold_words = pd.read_csv(refold_data, delimiter='\t', header=None)[1]
refold_words.head()

0    的
1    我
2    你
3    是
4    了
Name: 1, dtype: object

## heisig (characters, not words)
(~3000 chars)


In [37]:
heisig_words1 = pd.read_csv(heisig_data1, delimiter='\t')['Hanzi']
print(heisig_words1.head())
print(len(heisig_words1),'\n\n')
heisig_words2 = pd.read_csv(heisig_data2, delimiter='\t')['Hanzi']
print(heisig_words2.head())
print(len(heisig_words2),'\n\n')
heisig_words = pd.concat([heisig_words1,heisig_words2]).reset_index()['Hanzi']
heisig_words.head()
print(len(heisig_words))

0    一
1    二
2    三
3    四
4    五
Name: Hanzi, dtype: object
1500 


0    叭
1    咕
2    咱
3    串
4    罩
Name: Hanzi, dtype: object
2220 


3720


## character frequency (char. not words)
(10,000 chars)



In [34]:
char_freq = pd.read_csv(frequency_data, delimiter='\t', header=5)['汉字']
char_freq.head()

0       的
1       一
2       是
3       不
4       了
       ..
9928    鱀
9929    鱄
9930    鱓
9931    鲖
9932    鴒
Name: 汉字, Length: 9933, dtype: object

## Wiki (word frequency)
2 sets of ~1000,so ~2000 (but more can be added)

In [8]:
wiki_freq1_dat = "https://en.wiktionary.org/wiki/Appendix:Mandarin_Frequency_lists/1-1000"
wiki_freq1 = pd.read_html(wiki_freq1_dat)[0].iloc[:,:2].dropna()['Simplified']#[['Traditional','Simiplified','Pinyin']]
wiki_freq1.head()

0    一
2    在
4    有
5    个
7    我
Name: Simplified, dtype: object

In [9]:
wiki_freq2_dat = "https://en.wiktionary.org/wiki/Appendix:Mandarin_Frequency_lists/1001-2000"
wiki_freq2 = pd.read_html(wiki_freq2_dat)[0].iloc[:,:2].dropna()['Simplified']#[['Traditional','Simiplified','Pinyin']]
wiki_freq2.head()

0    回家
1    独立
2    英国
3    投资
4    理论
Name: Simplified, dtype: object

In [10]:
word_frequency = pd.concat([wiki_freq1,wiki_freq2]).reset_index()['Simplified']
word_frequency.head()

0    一
1    在
2    有
3    个
4    我
Name: Simplified, dtype: object

#Percent overlap

In [11]:
def basic_percent(source_list, master_list):
  total = 0
  occurence = 0
  for i,source in enumerate(source_list):
    # if i % 100 == 0:
    #   print(source)
    total += 1
    if source in master_list:
      occurence +=1
  print('percent similar:', occurence/total * 100)
  print(total)
  print(occurence)

basic_percent(list(refold_words)[:1000], list(word_frequency)[:1000])

percent similar: 45.5
1000
455


In [56]:
basic_percent(list(heisig_words1)[:], list(char_freq)[:1500])

percent similar: 80.46666666666667
1500
1207


- Heisig1 80.47% similarity to char frequency (equal size 1500:15000)
- Heisig 1 + 2 90.03% similar to char frequency list (equal size 3720:3720)